tests/src/main/java/com/hazelcast/simulator/tests/icache/AddRemoveListenerICacheTest.java

```diff
@@ -10,59 +10,65 @@
 import com.hazelcast.logging.ILogger;
 import com.hazelcast.logging.Logger;
 import com.hazelcast.simulator.test.TestContext;
-import com.hazelcast.simulator.test.annotations.Run;
+import com.hazelcast.simulator.test.TestRunner;
+import com.hazelcast.simulator.test.annotations.RunWithWorker;
 import com.hazelcast.simulator.test.annotations.Setup;
 import com.hazelcast.simulator.test.annotations.Verify;
 import com.hazelcast.simulator.test.annotations.Warmup;
-import com.hazelcast.simulator.tests.icache.helpers.MyCacheEntryEventFilter;
-import com.hazelcast.simulator.tests.icache.helpers.MyCacheEntryListener;
-import com.hazelcast.simulator.utils.ThreadSpawner;
+import com.hazelcast.simulator.tests.icache.helpers.ICacheEntryEventFilter;
+import com.hazelcast.simulator.tests.icache.helpers.ICacheEntryListener;
+import com.hazelcast.simulator.tests.icache.helpers.ICacheListenerOperationCounter;
+import com.hazelcast.simulator.worker.selector.OperationSelectorBuilder;
+import com.hazelcast.simulator.worker.tasks.AbstractWorker;
 
 import javax.cache.Cache;
 import javax.cache.CacheManager;
 import javax.cache.configuration.FactoryBuilder;
 import javax.cache.configuration.MutableCacheEntryListenerConfiguration;
-import java.io.Serializable;
-import java.util.Random;
 
 import static com.hazelcast.simulator.tests.helpers.HazelcastTestUtils.isMemberNode;
 
 /**
- * In This test we concurrently add remove cache listeners while putting and getting from the cache
- * this test is out side of normal usage, however has found problems where put operations hang
- * this type of test could uncover memory leaks in the process of adding and removing listeners
- * The max size of the cache used in this test is keyCount int key/value pairs,
+ * In this test we concurrently add remove cache listeners while putting and getting from the cache.
+ *
+ * This test is out side of normal usage, however has found problems where put operations hang.
+ * This type of test could uncover memory leaks in the process of adding and removing listeners.
+ * The max size of the cache used in this test is keyCount int key/value pairs.
  */
 public class AddRemoveListenerICacheTest {
 
+    private enum Operation {
+        REGISTER,
+        DE_REGISTER,
+        PUT,
+        GET
+    }
+
     private static final ILogger LOGGER = Logger.getLogger(AddRemoveListenerICacheTest.class);
 
-    public int threadCount = 3;
+    public String basename = AddRemoveListenerICacheTest.class.getSimpleName();
     public int keyCount = 1000;
     public boolean syncEvents = true;
 
-    public double register = 0;
-    public double deregister = 0;
-    public double put = 0;
-    public double get = 0;
+    public double registerProb = 0.25;
```

Comment:
Normally I search for the class name and adjust the `x.properties` files accordingly. I didn't find this class anywhere though.